contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,339 @@
+
+abstract contract BasePSM is IBasePSM, OracleRef, PCVDeposit {
+    using Decimal for Decimal.D256;
+    using SafeCast for *;
+    using SafeERC20 for IERC20;
+
+    /// @notice the amount of reserves to be held for redemptions
+    uint256 public override reservesThreshold;
+
+    /// @notice the PCV deposit target
+    IPCVDeposit public override surplusTarget;
+
+    /// @notice the token this PSM will exchange for VOLT
+    IERC20 public immutable override underlyingToken;
+
+    /// @notice struct for passing constructor parameters related to OracleRef
+    struct OracleParams {
+        address coreAddress;
+        address oracleAddress;
+        address backupOracle;
+        int256 decimalsNormalizer;
+        bool doInvert;
+    }
+
+    /// @notice constructor
+    /// @param params PSM constructor parameter struct
+    constructor(
+        OracleParams memory params,
+        uint256 _reservesThreshold,
+        IERC20 _underlyingToken,
+        IPCVDeposit _surplusTarget
+    )
+        OracleRef(
+            params.coreAddress,
+            params.oracleAddress,
+            params.backupOracle,
+            params.decimalsNormalizer,
+            params.doInvert
+        )
+    {
+        underlyingToken = _underlyingToken;
+
+        _setReservesThreshold(_reservesThreshold);
+        _setSurplusTarget(_surplusTarget);
+        _setContractAdminRole(keccak256("PSM_ADMIN_ROLE"));
+    }
+
+    /// @notice withdraw assets from PSM to an external address
+    function withdraw(address to, uint256 amount)
+        external
+        virtual
+        override
+        onlyPCVController
+    {
+        _withdrawERC20(address(underlyingToken), to, amount);
+    }
+
+    /// @notice set the ideal amount of reserves for the contract to hold for redemptions
+    function setReservesThreshold(uint256 newReservesThreshold)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setReservesThreshold(newReservesThreshold);
+    }
+
+    /// @notice set the target for sending surplus reserves
+    function setSurplusTarget(IPCVDeposit newSurplusTarget)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setSurplusTarget(newSurplusTarget);
+    }
+
+    /// @notice helper function to set reserves threshold
+    function _setReservesThreshold(uint256 newReservesThreshold) internal {
+        require(
+            newReservesThreshold > 0,
+            "PegStabilityModule: Invalid new reserves threshold"
+        );
+        uint256 oldReservesThreshold = reservesThreshold;
+        reservesThreshold = newReservesThreshold;
+
+        emit ReservesThresholdUpdate(
+            oldReservesThreshold,
+            newReservesThreshold
+        );
+    }
+
+    /// @notice helper function to set the surplus target
+    function _setSurplusTarget(IPCVDeposit newSurplusTarget) internal {
+        require(
+            address(newSurplusTarget) != address(0),
```

Comment:
This is a personal bug bear across Solidity, but this `!= address(0)` check isn't needed imo. 

One of the reasons this check became popular was because the compiler didn't use to check the length of `msg.data`, so missing args could be passed to functions. https://forum.openzeppelin.com/t/removing-address-0x0-checks-from-openzeppelin-contracts/2222 

If checking `!= address(0)`, why not also `!= address(1)` etc

Reply:
I would instead probably check that this new address isn't the same as the previous existing address